src/SpecialPage/SpecialResourceLoaderArticles.php

```diff
@@ -189,7 +189,7 @@ public function addPageCB( $formData ) {
 				. '</div>'
 			);
 			$store = false;
-		} elseif ( !is_int( $formData[ 'Priority' ] ) ) {
+		} elseif ( !ctype_digit( $formData[ 'Priority' ] ) ) {
```

Comment:
suggestion: `is_numeric` instead of `ctype_digit` https://www.php.net/manual/en/function.is-numeric.php

Reply:
I tried and we cannot enter a value in decimals since the field only accepts integers
![image](https://github.com/Liquipedia/ResourceLoaderArticles/assets/43470991/5890111f-d396-443c-aa21-31111131c0a0)

Reply:
The priority field still needs a string to int conversion because the value is received as a string `string(1) "2"`

Reply:
You might want to validate a max or a min, or you let the DB eat it. You don't really need to validate the return being numeric. string(1) means a string with length 1

Reply:
> So you agree we can drop that whole if statement, I think? But what about the checks for empty? It appears to me that they're not needed either if the form refuses it by itself without ever even reaching our manual if statements.

Yeah we haven't always relied on the tools given, some of these forms originally used to be just raw html and needed the validation, then were ported to HTMLForm, but the validation code might not have been updated.

Reply:
The empty checks should be kept but combined with "trim" because " " is still a valid string and is not empty.